### PR TITLE
first PR to depricate ansible

### PR DESCRIPTION
**Pull request by arturrez**

### CODE REVIEW

**Comment by felipemadero** [comment]
> > keep on getting this error
> > ```
> > 
> > New EC2 instance(s) successfully created in AWS!
> > Error: json: cannot unmarshal object into Go struct field ClustersConfigV0.Clusters of type []string
> > ```
> 
> This happens in main too, so not related to this PR

plase if having this issue, it is due to early testing of new cluster config file. add a field "Version" set to "1" to that file

**Comment by felipemadero** [comment]
> we keep on initializing terraform everytime we run create command, can we only run it once and not run it on subsequent runs?

we can start with this: https://github.com/ava-labs/avalanche-cli/issues/1147

and we can also add a issue for that

File: cmd/nodecmd/create.go
Comment (arturrez) [reply cmd/nodecmd/create.go:271]: we should I guess. Not sure how, open for the suggestions :)

File: cmd/nodecmd/create.go
Comment (arturrez) [reply cmd/nodecmd/create.go:292]: added error

File: cmd/nodecmd/create.go
Comment (arturrez) [reply cmd/nodecmd/create.go:289]: I removed a header and added error message, but I feel like we need to report user back it's instance IDs

File: cmd/nodecmd/create.go
Comment (felipemadero) [inline cmd/nodecmd/create.go:239]: is "to provision" clear regarding not beign able to connect to ssh?

File: cmd/nodecmd/create.go
Comment (felipemadero) [inline cmd/nodecmd/create.go:257]: all of this to use the same ssh connection on host. end with a new host.CloseConn function call

File: pkg/ssh/ssh.go
Comment (felipemadero) [inline pkg/ssh/ssh.go:99]: there is some \n\n double observable in the output, that should be avoided in this code

File: cmd/nodecmd/create.go
Comment (arturrez) [reply cmd/nodecmd/create.go:252]: fixed

File: cmd/nodecmd/create.go
Comment (arturrez) [reply cmd/nodecmd/create.go:251]: map with lock is a possibility, I agree

Comment (arturrez) [reply cmd/nodecmd/create.go:251]: but it will require another struct

File: cmd/nodecmd/create.go
Comment (arturrez) [reply cmd/nodecmd/create.go:239]: err will have this information

File: pkg/models/misc.go
Comment (arturrez) [reply pkg/models/misc.go:5]: it will be in the following PRs

File: pkg/models/misc.go
Comment (arturrez) [reply pkg/models/misc.go:16]: it will be in the following PR

File: cmd/nodecmd/create.go
Comment (arturrez) [reply cmd/nodecmd/create.go:251]: let me do it

File: cmd/nodecmd/create.go
Comment (arturrez) [reply cmd/nodecmd/create.go:257]: goph doesn't let me reuse the connection unfortunatelly

File: pkg/ansible/ansible.go
Comment (arturrez) [reply pkg/ansible/ansible.go:126]: wdym?

File: pkg/constants/constants.go
Comment (arturrez) [reply pkg/constants/constants.go:50]: we also copy files

File: pkg/constants/constants.go
Comment (arturrez) [reply pkg/constants/constants.go:49]: reduced to 120

File: pkg/constants/constants.go
Comment (arturrez) [reply pkg/constants/constants.go:47]: bad merge, sorry fixed

File: pkg/models/host.go
Comment (arturrez) [reply pkg/models/host.go:251]: didn't find any :(

File: pkg/models/host.go
Comment (arturrez) [reply pkg/models/host.go:240]: fixed

File: pkg/models/host.go
Comment (arturrez) [reply pkg/models/host.go:123]: wdym?

File: pkg/models/host.go
Comment (arturrez) [reply pkg/models/host.go:216]: good one. fixed

File: pkg/models/host.go
Comment (arturrez) [reply pkg/models/host.go:218]: agree

File: pkg/models/host.go
Comment (arturrez) [reply pkg/models/host.go:172]: we don't need this func - removed

File: pkg/models/host.go
Comment (arturrez) [reply pkg/models/host.go:165]: good one

File: pkg/models/host.go
Comment (arturrez) [reply pkg/models/host.go:32]: yes

File: pkg/models/host.go
Comment (arturrez) [reply pkg/models/host.go:87]: I can but it will waste of time if folder exists and waste of time to check if folder exists for each upload ( making things slower), so I leave it to user to make sure he knows what he is doing

Comment (arturrez) [reply pkg/models/host.go:87]: fixed

File: pkg/ssh/shell/setupBuildEnv.sh
Comment (arturrez) [reply pkg/ssh/shell/setupBuildEnv.sh:1]: it's minor, we should do it later if we want to

File: pkg/ssh/shell/setupBuildEnv.sh
Comment (arturrez) [reply pkg/ssh/shell/setupBuildEnv.sh:2]: sometimes it's not enough and it's recommented to let apt know via env var

File: pkg/utils/ssh.go
Comment (arturrez) [reply pkg/utils/ssh.go:12]: wdym?

File: pkg/ssh/shell/setupNode.sh
Comment (arturrez) [reply pkg/ssh/shell/setupNode.sh:11]: no positional args by design

File: pkg/ssh/shell/trackSubnet.sh
Comment (arturrez) [reply pkg/ssh/shell/trackSubnet.sh:6]: we should open an issue for this as this code is on par with ansible implementation

File: pkg/ssh/ssh.go
Comment (arturrez) [reply pkg/ssh/ssh.go:87]: disagree

File: pkg/ssh/ssh.go
Comment (arturrez) [reply pkg/ssh/ssh.go:99]: good one

Comment (arturrez) [reply pkg/ssh/ssh.go:99]: fixed

File: pkg/ssh/ssh.go
Comment (arturrez) [reply pkg/ssh/ssh.go:103]: just trying to do single retry just in case

File: pkg/ssh/ssh.go
Comment (arturrez) [reply pkg/ssh/ssh.go:22]: yes, for the template to work. unless I am mistaken

File: pkg/ssh/ssh.go
Comment (arturrez) [reply pkg/ssh/ssh.go:38]: we should give users some idea what's happening as it will have to wait long

File: pkg/constants/constants.go
Comment (felipemadero) [reply pkg/constants/constants.go:152]: this is not used and was not removed

File: pkg/utils/common.go
Comment (felipemadero) [reply pkg/utils/common.go:34]: could you please remove this?

File: pkg/utils/common.go
Comment (felipemadero) [reply pkg/utils/common.go:123]: could you remove this?

File: pkg/ansible/ansible.go
Comment (felipemadero) [reply pkg/ansible/ansible.go:127]: you can use utils.Map to replace the outermost iteration

File: pkg/models/host.go
Comment (felipemadero) [reply pkg/models/host.go:52]: there was no constant set here: you can use SSHTCPPort

File: pkg/models/host.go
Comment (felipemadero) [reply pkg/models/host.go:56]: no comment added. which is the purpose of the field? used as a callaback?

File: pkg/ssh/ssh.go
Comment (felipemadero) [reply pkg/ssh/ssh.go:67]: this does not need to be exported

File: pkg/ssh/ssh.go
Comment (felipemadero) [reply pkg/ssh/ssh.go:74]: no constant added

File: pkg/ssh/ssh.go
Comment (felipemadero) [reply pkg/ssh/ssh.go]: no constant added

File: pkg/ssh/shell/setupNode.sh
Comment (felipemadero) [inline pkg/ssh/shell/setupNode.sh:2]: I believe we don't need this annotations anymore. Same for the other scripts

File: pkg/models/host.go
Comment (arturrez) [reply pkg/models/host.go:56]: adding comment

File: pkg/models/host.go
Comment (felipemadero) [inline pkg/models/host.go:21]: I believe you changed the number by mistake. Could you add a comment on why you choose this number?

File: pkg/ssh/shell/cleanupDevnet.sh
Comment (felipemadero) [inline pkg/ssh/shell/cleanupDevnet.sh:8]: "shell: " should not be here

File: pkg/ssh/shell/cleanupDevnet.sh
Comment (felipemadero) [inline pkg/ssh/shell/cleanupDevnet.sh:1]: can we call this setupDevnet.sh or startDevnet.sh or initDevnet.sh

File: cmd/nodecmd/create_devnet.go
Comment (felipemadero) [inline cmd/nodecmd/create_devnet.go:225]: can we stop configuration and return error is some node was not setup?

File: pkg/ssh/shell/setupNode.sh
Comment (felipemadero) [inline pkg/ssh/shell/setupNode.sh:18]: can we stop the node here if this is a devnet?

File: pkg/ssh/shell/cleanupDevnet.sh
Comment (arturrez) [reply pkg/ssh/shell/cleanupDevnet.sh:8]: yes this is correct. bug
fixed

File: pkg/ssh/shell/setupNode.sh
Comment (arturrez) [reply pkg/ssh/shell/setupNode.sh:2]: this is comment in the script. We can leave it even if don't display it to the users

File: pkg/models/host.go
Comment (arturrez) [reply pkg/models/host.go:21]: this is just max size of byte slice allocated to the response to be parsed.
it can be any number that's why I made it bigger

Comment (arturrez) [reply pkg/models/host.go:21]: added comment

File: pkg/ssh/shell/cleanupDevnet.sh
Comment (arturrez) [reply pkg/ssh/shell/cleanupDevnet.sh]: it was the cause of devnet issue.

File: pkg/ssh/shell/cleanupDevnet.sh
Comment (arturrez) [reply pkg/ssh/shell/cleanupDevnet.sh]: done

File: cmd/nodecmd/create_devnet.go
Comment (arturrez) [reply cmd/nodecmd/create_devnet.go:225]: yes

**Review by felipemadero — APPROVED** [review]
LGTM this also needs @sukantoraymond approval

**Pull request merged**
